### PR TITLE
Ddf 2401 added Plugins section to revised docs

### PR DESCRIPTION
**Pull request by ricklarsen**
#### What does this PR do?

Adds "Plugins" to Integrating DDF docs. 
#### Who is reviewing it?

@brendan-hofmann @ahoffer @AzGoalie
#### Choose 2 committers to review/merge the PR.

@michaelmenousek 
@pklinef
@shaundmorris 
#### How should this be tested?

docs render correctly, build passes, info complete & up-to-date.
#### Any background context you want to provide?

Sub-task of https://codice.atlassian.net/browse/DDF-2309. 
Some sections exist only as template placeholder and will need to either be completed or deleted.
#### What are the relevant tickets?

https://codice.atlassian.net/browse/DDF-2309
https://codice.atlassian.net/browse/DDF-2401
#### Screenshots (if appropriate)

[draft-documentation.pdf](https://github.com/codice/ddf/files/460042/draft-documentation.pdf)
#### Checklist:
- [x] Documentation Updated
- [ ] Update / Add Unit Tests
- [ ] Update / Add Integration Tests

### CODE REVIEW

**Comment by ricklarsen** [comment]
courtesy tag: @mcalcote

File: distribution/docs/src/main/resources/_contents/_plugins/identification-plugin-contents.adoc [hunk 1]
```diff
@@ -0,0 +1,15 @@
+==== Identification Plugin
+
+The Identification Plugin assigns IDs to a metacard as it is added to the catalog.
+
+===== Installing Identification Plugin
+
+The Identification Plugin is is not installed by default in a standard installation in the ${ddf-registry} application.
```
Comment (mcalcote) [inline distribution/docs/src/main/resources/_contents/_plugins/identification-plugin-contents.adoc]: It is installed by default when starting the DDF Registry Application

File: distribution/docs/src/main/resources/_contents/_plugins/identification-plugin-contents.adoc [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+
+===== Configuring the Identification Plugin
+
+{configuration}
```
Comment (mcalcote) [inline distribution/docs/src/main/resources/_contents/_plugins/identification-plugin-contents.adoc]: No Config and no known issues

File: distribution/docs/src/main/resources/_contents/_plugins/registry-policy-plugin-contents.adoc [hunk 3]
```diff
@@ -0,0 +1,17 @@
+
+==== Registry Policy Plugin
+
+{description}
+
+===== Installing Registry Policy Plugin
+
+The Registry Policy Plugin is is not installed by default with a standard installation in the ${app-name} application.
```
Comment (mcalcote) [inline distribution/docs/src/main/resources/_contents/_plugins/registry-policy-plugin-contents.adoc]: Is installed by default when the DDF-Registry App is started

File: distribution/docs/src/main/resources/_contents/_plugins/registry-policy-plugin-contents.adoc [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+
+===== Configuring Registry Policy Plugin
+
+{configuration}
```
Comment (mcalcote) [inline distribution/docs/src/main/resources/_contents/_plugins/registry-policy-plugin-contents.adoc]: . Navigate to the ${admin-console}.
. Select the *${ddf-registry}* application tile.
. Select the *Configuration* tab.
. Select the *Registry Policy Plugin* configuration.
 +
 +include::{adoc-include}/_tables/ ... policyplugin adoc link here

File: distribution/docs/src/main/resources/_contents/_plugins/developing-plugins-contents.adoc [hunk 5]
```diff
@@ -0,0 +1,378 @@
...
+|Access
+|${ddf-branding}.catalog.plugin.AccessPlugin
+|Runs directly after the PolicyPlugin
+|An examples are the FilterPlugin and OperationPlugin
+
+|===
+
+==== Implement Plugins
+
+The procedure for implementing any of the plugins follows a similar format:
```
Comment (brendan-hofmann) [inline distribution/docs/src/main/resources/_contents/_plugins/developing-plugins-contents.adoc]: Might want to add a note about performance and threading concerns. One bad plugin can tank the system.

File: distribution/docs/src/main/resources/_contents/_plugins/jpeg2000-thumbnail-converter-contents.adoc [hunk 6]
```diff
@@ -0,0 +1,16 @@
+==== JPEG2000 Thumbnail Converter
+
+{description}
+
+===== Installing JPEG2000 Thumbnail Converter
+
+The JPEG2000 Thumbnail Converter is is not installed by default with a standard installation in the ${app-name} application.
```
Comment (brendan-hofmann) [inline distribution/docs/src/main/resources/_contents/_plugins/jpeg2000-thumbnail-converter-contents.adoc]: Yes it is. Also, typo 'is is' in all of these.

File: distribution/docs/src/main/resources/_contents/_plugins/policy-plugin-intro-contents.adoc [hunk 7]
```diff
@@ -0,0 +1,5 @@
+
+=== Policy Plugins
```
Comment (brendan-hofmann) [inline distribution/docs/src/main/resources/_contents/_plugins/policy-plugin-intro-contents.adoc]: Shouldn't we cover this under security also/instead?

File: distribution/docs/src/main/resources/_contents/_plugins/_plugin-contents-template.adoc [hunk 8]
```diff
@@ -0,0 +1,17 @@
+
+==== {plugin-name}
+
+{description}
+
+===== Installing the {plugin-name}
+
+The {plugin-name} is is not installed by default with a standard installation in the ${app-name} application.
```
Comment (AzGoalie) [inline distribution/docs/src/main/resources/_contents/_plugins/_plugin-contents-template.adoc]: double is?

File: distribution/docs/src/main/resources/_contents/_plugins/identification-plugin-contents.adoc (hunk 1 above)
Comment (ricklarsen) [inline distribution/docs/src/main/resources/_contents/_plugins/identification-plugin-contents.adoc]: fixed.

File: distribution/docs/src/main/resources/_contents/_plugins/identification-plugin-contents.adoc (hunk 2 above)
Comment (ricklarsen) [inline distribution/docs/src/main/resources/_contents/_plugins/identification-plugin-contents.adoc]: fixed.

File: distribution/docs/src/main/resources/_contents/_plugins/registry-policy-plugin-contents.adoc (hunk 4 above)
Comment (ricklarsen) [inline distribution/docs/src/main/resources/_contents/_plugins/registry-policy-plugin-contents.adoc]: added.

File: distribution/docs/src/main/resources/_contents/_plugins/developing-plugins-contents.adoc (hunk 5 above)
Comment (ricklarsen) [inline distribution/docs/src/main/resources/_contents/_plugins/developing-plugins-contents.adoc]: I could agree to both of these. Let me know what you want to say

File: distribution/docs/src/main/resources/_contents/_plugins/jpeg2000-thumbnail-converter-contents.adoc (hunk 6 above)
Comment (ricklarsen) [inline distribution/docs/src/main/resources/_contents/_plugins/jpeg2000-thumbnail-converter-contents.adoc]: not quite a typo, looking for people to confirm `is` or `is not`

File: distribution/docs/src/main/resources/_contents/_plugins/policy-plugin-intro-contents.adoc (hunk 7 above)
Comment (ricklarsen) [inline distribution/docs/src/main/resources/_contents/_plugins/policy-plugin-intro-contents.adoc]: I'd rather link to here from there.

File: distribution/docs/src/main/resources/_contents/_plugins/developing-plugins-contents.adoc [hunk 9]
```diff
@@ -0,0 +1,378 @@
+
+=== Developing Catalog Plugins
+
+Plugins extend the functionality of the Catalog Framework by performing actions at specified times during a transaction.
```
Comment (shaundmorris) [inline distribution/docs/src/main/resources/_contents/_plugins/developing-plugins-contents.adoc]: Should we add a note that these interfaces are located in the catalog core api?

File: distribution/docs/src/main/resources/_contents/_plugins/developing-plugins-contents.adoc (hunk 5 above)
Comment (ricklarsen) [inline distribution/docs/src/main/resources/_contents/_plugins/developing-plugins-contents.adoc]: I'll check in with @brendan-hofmann today.

File: distribution/docs/src/main/resources/_contents/_plugins/developing-plugins-contents.adoc (hunk 9 above)
Comment (ricklarsen) [inline distribution/docs/src/main/resources/_contents/_plugins/developing-plugins-contents.adoc]: sure.

File: distribution/docs/src/main/resources/_contents/_plugins/fanout-event-processor-contents.adoc [hunk 10]
```diff
@@ -0,0 +1,19 @@
+==== Fanout Event Processor
```
Comment (ricklarsen) [inline distribution/docs/src/main/resources/_contents/_plugins/fanout-event-processor-contents.adoc]: I'll look into it.

File: distribution/docs/src/main/resources/_contents/_plugins/metacard-validity-filter-plugin-contents.adoc [hunk 11]
```diff
@@ -0,0 +1,24 @@
+
+==== Metacard Validation Filter Plugin
+
+The Metacard Validation Filter Plugin maps the attributes between user and metacard and determines whether metacards with validation errors or warnings are filtered from query results.
```
Comment (kcwire) [inline distribution/docs/src/main/resources/_contents/_plugins/metacard-validity-filter-plugin-contents.adoc]: This seems wrong - I don't think this is mapping user & metacard.

File: distribution/docs/src/main/resources/_contents/_plugins/historian-policy-plugin-contents.adoc [hunk 12]
```diff
@@ -0,0 +1,19 @@
+
+==== Historian Plugin
```
Comment (rzwiefel) [inline distribution/docs/src/main/resources/_contents/_plugins/historian-policy-plugin-contents.adoc]: So technically there is no longer a Historian "plugin".  Well there is and isn't.

Now, the Historian is embedded in the catalog framework and is configurable via a toggle in the catalog admin config. There is also still a Historian Policy plugin that is always enabled, which allows only users with the `system-history` role to create/modify/delete history items.

File: distribution/docs/src/main/resources/_contents/_tables/ddf.catalog.registry.policy.RegistryPolicyPlugin-table-contents.adoc [hunk 13]
```diff
@@ -0,0 +1,55 @@
...
+|===
+
+|Name
+|Id
+|Type
+|Description
+|Default Value
+|Required
+
+|Registry CUD Attributes
```
Comment (clockard) [inline distribution/docs/src/main/resources/_contents/_tables/ddf.catalog.registry.policy.RegistryPolicyPlugin-table-contents.adoc]: This field has been broken up into three separate ones so you can controll create/update/delete individually.
Here is the info from the metatype
```
<AD description="Roles/attributes required for create operations on registry entries. Example: {role=role1;type=type1}"
            name="Registry Create Attributes" id="createAccessPolicyStrings" required="true"
            type="String" cardinality="100"
            default="http://schemas.xmlsoap.org/ws/2005/05/identity/claims/role=guest"/>

        <AD description="Roles/attributes required for update operations on registry entries. Example: {role=role1;type=type1}"
            name="Registry Update Attributes" id="updateAccessPolicyStrings" required="true"
            type="String" cardinality="100"
            default="http://schemas.xmlsoap.org/ws/2005/05/identity/claims/role=guest"/>

        <AD description="Roles/attributes required for delete operations on registry entries. Example: {role=role1;type=type1}"
            name="Registry Delete Attributes" id="deleteAccessPolicyStrings" required="true"
            type="String" cardinality="100"
            default="http://schemas.xmlsoap.org/ws/2005/05/identity/claims/role=guest"/>
```

File: distribution/docs/src/main/resources/_contents/_tables/ddf.catalog.registry.policy.RegistryPolicyPlugin-table-contents.adoc [hunk 14]
```diff
@@ -0,0 +1,55 @@
+.[[ddf.catalog.registry.policy.RegistryPolicyPlugin]]Registry Policy Plugin
```
Comment (clockard) [inline distribution/docs/src/main/resources/_contents/_tables/ddf.catalog.registry.policy.RegistryPolicyPlugin-table-contents.adoc]: Just realized that this file should be replaced with the `org.codice.ddf.catalog.registry.policy.RegistryPolicyPlugin-table-contents.adoc`

File: distribution/docs/src/main/resources/_contents/_tables/org.codice.ddf.registry.policy.RegistryPolicyPlugin-table-contents.adoc [hunk 15]
```diff
@@ -0,0 +1,55 @@
...
+|===
+
+|Name
+|Id
+|Type
+|Description
+|Default Value
+|Required
+
+|Registry CUD Attributes
```
Comment (clockard) [inline distribution/docs/src/main/resources/_contents/_tables/org.codice.ddf.registry.policy.RegistryPolicyPlugin-table-contents.adoc]: See comment on outdated registry policy plugin file

File: distribution/docs/src/main/resources/_contents/_tables/org.codice.ddf.registry.policy.RegistryPolicyPlugin-table-contents.adoc [hunk 16]
```diff
@@ -0,0 +1,55 @@
+.[[org.codice.ddf.registry.policy.RegistryPolicyPlugin]]Registry Policy Plugin
```
Comment (clockard) [inline distribution/docs/src/main/resources/_contents/_tables/org.codice.ddf.registry.policy.RegistryPolicyPlugin-table-contents.adoc]: This is the correct RegistryPolicyPlugin file.

File: distribution/docs/src/main/resources/_contents/_plugins/developing-plugins-contents.adoc (hunk 5 above)
Comment (ricklarsen) [inline distribution/docs/src/main/resources/_contents/_plugins/developing-plugins-contents.adoc]: added.

File: distribution/docs/src/main/resources/_contents/_plugins/developing-plugins-contents.adoc [hunk 17]
```diff
@@ -0,0 +1,383 @@
...
+
+The procedure for implementing any of the plugins follows a similar format:
+
+. Create a new class that implements the specified plugin interface.
+
+. Implement the required methods.
+
+. Create OSGi descriptor file to communicate with the OSGi registry.
+
+.. Import ${branding} packages.
```
Comment (ricklarsen) [inline distribution/docs/src/main/resources/_contents/_plugins/developing-plugins-contents.adoc]: fixed.

File: distribution/docs/src/main/resources/_contents/_plugins/developing-plugins-contents.adoc [hunk 18]
```diff
@@ -0,0 +1,383 @@
...
+
+===== Implementing Pre-Ingest Plugins
+
+Develop a custom Pre-Ingest Plugin.
+
+. Create a Java class that implements `PreIngestPlugin.` +
+`public class SamplePreIngestPlugin implements ddf.catalog.plugin.PreIngestPlugin`
++
+. Implement the required methods. +
+ * `public CreateRequest process(CreateRequest input) throws PluginExecutionException;`
```
Comment (ricklarsen) [inline distribution/docs/src/main/resources/_contents/_plugins/developing-plugins-contents.adoc]: added

File: distribution/docs/src/main/resources/_contents/_plugins/developing-plugins-contents.adoc [hunk 19]
```diff
@@ -0,0 +1,383 @@
...
+*Blueprint descriptor example*
+`<service ref="[[SamplePostIngestPlugin ]]"interface="ddf.catalog.plugin.PostIngestPlugin" />`
+
+====== Post-Ingest Plugin Failure Behavior
+
+In the event that this Catalog Plugin cannot operate but does not wish to fail the transaction, a `PluginExecutionException` will be thrown.
+
+====== Invoking a Post-Ingest Plugin
+
+Because the event has already occurred and changes from one post-ingest plugin cannot affect others, all Post-Ingest plugins are invoked in parallel and no priority is enforced.
```
Comment (ricklarsen) [inline distribution/docs/src/main/resources/_contents/_plugins/developing-plugins-contents.adoc]: fixed.

File: distribution/docs/src/main/resources/_contents/_plugins/metacard-groomer-contents.adoc [hunk 20]
```diff
@@ -0,0 +1,25 @@
+==== Metacard Groomer
+
+The Metacard Groomer Pre-Ingest plugin makes modifications to `CreateRequest` and `UpdateRequest` metacards.
+
+Use this pre-ingest plugin as a convenience to apply basic rules for your metacards. 
+
+This plugin makes the following modifications when metacards are in a `CreateRequest`:
+
+* Overwrites the `Metacard.ID` field with a generated, unique, 32 character hexadecimal value.
```
Comment (ricklarsen) [inline distribution/docs/src/main/resources/_contents/_plugins/metacard-groomer-contents.adoc]: fixed.

File: distribution/docs/src/main/resources/_contents/_plugins/metacard-groomer-contents.adoc [hunk 21]
```diff
@@ -0,0 +1,25 @@
...
+* Overwrites the `Metacard.CREATED` date with a current time stamp.
+* Overwrites the `Metacard.MODIFIED` date with a current time stamp.
+
```
Comment (ricklarsen) [inline distribution/docs/src/main/resources/_contents/_plugins/metacard-groomer-contents.adoc]: fixed

File: distribution/docs/src/main/resources/_contents/_plugins/metacard-groomer-contents.adoc [hunk 22]
```diff
@@ -0,0 +1,25 @@
...
+The plugin also makes the following modifications when metacards are in an `UpdateRequest`:
+
+* If no value is provided for `Metacard.ID` in the new metacard, it will be set using the `UpdateRequest` ID if applicable.
+* If no value is provided, sets the `Metacard.CREATED` date with the `Metacard.MODIFIED` date so that the `Metacard.CREATED` date is not null.
+* Overwrites the `Metacard.MODIFIED` date with a current time stamp.
+
+===== Installing the Metacard Groomer
+
+The Metacard Groomer is not installed by default with a standard installation in the ${ddf-catalog} application.
```
Comment (ricklarsen) [inline distribution/docs/src/main/resources/_contents/_plugins/metacard-groomer-contents.adoc]: fixed.

File: distribution/docs/src/main/resources/_contents/_plugins/metacard-resource-size-plugin-contents.adoc [hunk 23]
```diff
@@ -0,0 +1,13 @@
+==== Metacard Resource Size Plugin
+
+This post-query plugin updates the resource size attribute of each metacard in the query results if there is a cached file for the product and it has a size greater than zero; otherwise, the resource size is unmodified and the original result is returned.
+
+Use this post-query plugin as a convenience to return query results with accurate resource sizes for cached products. 
+
+===== Installing the Metacard Resource Size Plugin
+
+The Metacard Resource Size Plugin is not installed by default with a standard installation.
```
Comment (ricklarsen) [inline distribution/docs/src/main/resources/_contents/_plugins/metacard-resource-size-plugin-contents.adoc]: fixed.

File: distribution/docs/src/main/resources/_contents/_plugins/plugin-compatibility-table-contents.adoc [hunk 24]
```diff
@@ -0,0 +1,121 @@
...
+.${ddf-branding} "Post" Plugin Compatibility
+[cols="8" options="header"]
+|===
+|Plugin
+|<<_post_ingest_plugins,Post-Ingest>>
+|<<_post_query_plugins,Post-Query>>
+|<<_post_resource_plugins,Post-Resource>>
+|<<_post_create storage_plugins,Post-Create Storage>>
+|<<_post_update storage_plugins,Post-Update Storage>>
+|<<_policy_plugins,Policy>>
```
Comment (ricklarsen) [inline distribution/docs/src/main/resources/_contents/_plugins/plugin-compatibility-table-contents.adoc]: done.

File: distribution/docs/src/main/resources/_contents/_plugins/policy-plugin-intro-contents.adoc [hunk 25]
```diff
@@ -0,0 +1,7 @@
...
+
+Policy plugins are invoked before all other plugin types to set up the policy for a request/response.
```
Comment (ricklarsen) [inline distribution/docs/src/main/resources/_contents/_plugins/policy-plugin-intro-contents.adoc]: We've been working on it.

File: distribution/docs/src/main/resources/_contents/_plugins/post-ingest-plugins-contents.adoc [hunk 26]
```diff
@@ -0,0 +1,13 @@
...
+.[[_included_post-ingest_plugins]]Included Post-Ingest Plugins
+<<_catalog_backup_plugin,Catalog Backup Plugin>>:: Enables data backup of the catalog and the metacards it contains.
+
+<<_catalog_metrics_plugin,Catalog Metrics Plugin>>:: The Catalog Metrics Plugin captures metrics on catalog operations.
+
+<<_event_processor,Event Processor>>:: Creates, updates, and deletes subscriptions for event notification.
+
+<<_fanout_event_processor,Fanout Event Processor>>:: Creates, updates, and deletes subscriptions for event notification.
+
+<<_identification_plugin,Identification Plugin>>:: The Identification Plugin assigns IDs to a metacard as it is added to the catalog.
```
Comment (ricklarsen) [inline distribution/docs/src/main/resources/_contents/_plugins/post-ingest-plugins-contents.adoc]: fixed.

File: distribution/docs/src/main/resources/_contents/_tables/ddf.catalog.registry.policy.RegistryPolicyPlugin-table-contents.adoc (hunk 14 above)
Comment (ricklarsen) [inline distribution/docs/src/main/resources/_contents/_tables/ddf.catalog.registry.policy.RegistryPolicyPlugin-table-contents.adoc]: fixed.

File: distribution/docs/src/main/resources/_contents/_tables/ddf.catalog.registry.policy.RegistryPolicyPlugin-table-contents.adoc (hunk 13 above)
Comment (ricklarsen) [inline distribution/docs/src/main/resources/_contents/_tables/ddf.catalog.registry.policy.RegistryPolicyPlugin-table-contents.adoc]: fixed.

File: distribution/docs/src/main/resources/_contents/_tables/org.codice.ddf.registry.policy.RegistryPolicyPlugin-table-contents.adoc (hunk 16 above)
Comment (ricklarsen) [inline distribution/docs/src/main/resources/_contents/_tables/org.codice.ddf.registry.policy.RegistryPolicyPlugin-table-contents.adoc]: it's the one getting used in the Registry Policy Plugin page

File: distribution/docs/src/main/resources/_contents/_plugins/historian-policy-plugin-contents.adoc (hunk 12 above)
Comment (shaundmorris) [inline distribution/docs/src/main/resources/_contents/_plugins/historian-policy-plugin-contents.adoc]: @rzwiefel do you think we should remove this "historian plugin" section from the docs?

Comment (rzwiefel) [inline distribution/docs/src/main/resources/_contents/_plugins/historian-policy-plugin-contents.adoc]: Yup go for it :+1: Sorry didn't see that response. Github notifications. 🙃

Comment (ricklarsen) [inline distribution/docs/src/main/resources/_contents/_plugins/historian-policy-plugin-contents.adoc]: removed

**Pull request merged**